crates/bevy_pbr/src/meshlet/from_mesh.rs

```diff
@@ -0,0 +1,98 @@
+use super::asset::{Meshlet, MeshletBoundingSphere, MeshletMesh};
+use bevy_render::{
+    mesh::{Indices, Mesh},
+    render_resource::PrimitiveTopology,
+};
+use meshopt::{build_meshlets, compute_meshlet_bounds_decoder, VertexDataAdapter};
+use std::borrow::Cow;
+
+impl MeshletMesh {
+    /// Process a [`Mesh`] to generate a [`MeshletMesh`].
+    ///
+    /// This process is very slow, and should be done ahead of time, and not at runtime.
+    ///
+    /// This function requires the `meshlet_processor` cargo feature.
+    ///
+    /// The input mesh must:
+    /// 1. Use [`PrimitiveTopology::TriangleList`]
+    /// 2. Use indices
+    /// 3. Have the exact following set of vertex attributes: `{POSITION, NORMAL, UV_0, TANGENT}`
+    pub fn from_mesh(mesh: &Mesh) -> Result<Self, MeshToMeshletMeshConversionError> {
+        // Validate mesh format
+        if mesh.primitive_topology() != PrimitiveTopology::TriangleList {
+            return Err(MeshToMeshletMeshConversionError::WrongMeshPrimitiveTopology);
+        }
+        if mesh.attributes().map(|(id, _)| id).ne([
+            Mesh::ATTRIBUTE_POSITION.id,
+            Mesh::ATTRIBUTE_NORMAL.id,
+            Mesh::ATTRIBUTE_UV_0.id,
+            Mesh::ATTRIBUTE_TANGENT.id,
+        ]) {
```

Comment:
I feel like this could be made more forgiving by requiring these attributes, and discarding others. So maybe we could add a `get_vertex_buffer_data_from_attributes(attributes: &[...])` that gathers the attributes in the specified order into a vertex buffer.

Doesn't have to block, but that feels generally useful (we could use it for making a position-only buffer and an everything else buffer, for example).

Reply:
Sure, let's leave that to another PR though. The requirements are also going to change in the near future for meshlets specifically. I want to add compression, and experiment with omitting the normal/tangents entirely and computing them at runtime since we have such high poly geometry.